specifyweb/frontend/js_src/lib/components/WorkBench/WbSpreadsheet.tsx

```diff
@@ -298,144 +280,190 @@ function WbSpreadsheetComponent({
-                // typeof this.uploadedView === 'function' ||
-                // typeof this.coordinateConverterView === 'function' ||
-                !disambiguation.isAmbiguousCell() ||
-                !hasPermission('/workbench/dataset', 'update'),
+                !disambiguation.isAmbiguousCell() || isReadOnly,
               callback: () => openDisambiguationDialog(),
             },
             separator_1: '---------',
-            fill_down: fillCellsContextMenuItem(hotRef, 'down'),
-            fill_up: fillCellsContextMenuItem(hotRef, 'up'),
+            fill_down: fillCellsContextMenuItem(hot!, 'down', isReadOnly),
+            fill_up: fillCellsContextMenuItem(hot!, 'up', isReadOnly),
             separator_2: '---------',
             undo: {
-              disabled: () =>
-                // TODO: Figure out if these commented lines can be removed
-                // typeof this.uploadedView === 'function' ||
-                hotRef.current?.hotInstance!.isUndoAvailable() !== true ||
-                !hasPermission('/workbench/dataset', 'update'),
+              disabled: () => hot?.isUndoAvailable() !== true || isReadOnly,
             },
             redo: {
-              disabled: () =>
-                // TODO: Figure out if these commented lines can be removed
-                // typeof this.uploadedView === 'function' ||
-                hotRef.current?.hotInstance!.isRedoAvailable() !== true ||
-                !hasPermission('/workbench/dataset', 'update'),
+              disabled: () => hot?.isRedoAvailable() !== true || isReadOnly,
             },
           } as const)
     ),
   };
 
+  React.useEffect(() => {
+    const getPickLists = async () =>
+      mappings === undefined
+        ? {}
+        : await fetchWbPickLists(
+            dataset.columns,
+            mappings.tableNames,
+            mappings.lines
+          );
+    if (hot) {
+      const configureHotAndPickLists = async () => {
+        const pickLists = await getPickLists();
+        configureHandsontable(hot, mappings, dataset, pickLists);
+      };
+      configureHotAndPickLists();
+    }
+  }, [hot]);
+
   // Highlight validation cells
-  React.useLayoutEffect(() => {
-    // TODD: Verify if previous code does anything else after validation
-    if (dataset.rowresults && hotRef.current) {
+  React.useEffect(() => {
+    // TODO: Verify if previous code does anything else after validation
+    if (dataset.rowresults && hot) {
       validation.getValidationResults();
+      workbench.cells.flushIndexedCellData = true;
     }
-  }, [dataset.rowresults, hotRef.current])
+  }, [dataset.rowresults, hot]);
+
+  // TODO: Move hottable props to a different file?
+  const autoWrapCol = React.useMemo(
+    () => userPreferences.get('workBench', 'editor', 'autoWrapCol'),
+    []
+  );
+  const autoWrapRow = React.useMemo(
+    () => userPreferences.get('workBench', 'editor', 'autoWrapRow'),
+    []
+  );
```

Comment:
no need for useMemo as userPreferences.get is cheap. useMemo might be adding more overhead